internal/indexer/definitions/funfile.yaml

```diff
@@ -0,0 +1,72 @@
+---
+#id: funfile
+name: FunFile
+identifier: funfile
+description: Welcome to funfile.org, where files are fun!
+language: en-us
+urls:
+  - https://funfile.org
+privacy: private
+protocol: torrent
+supports:
+  - irc
+  - rss
+source: custom
+settings:
+  - name: passkey
+    type: secret
+    required: true
+    label: Passkey
+    help: "The passkey in your FunFile RSS feed link."
+
+irc:
+  network: FunFile
+  server: irc.funfile.org
+  port: 6697
+  tls: true
+  channels:
+    - "#FunFile.org"
+  announcers:
+    - H3r4ut
+  settings:
+    - name: nick
+      type: text
+      required: true
+      label: Nick
+      help: Use your site username.
+
+    - name: auth.account
+      type: text
+      required: true
+      label: NickServ Account
+      help: NickServ account. Make sure to group your user and bot.
+
+    - name: auth.password
+      type: secret
+      required: true
+      label: NickServ Password
+      help: NickServ password
+
+  parse:
+    type: multi
+    lines:
+      - test:
+          - '[New Shit] S.W.A.T.Firefight.2011.PROPER.NTSC.MULTi.DVDR-BonG [Movies] @ https://www.funfile.org/details.php?id=131267 [size] [freeleech]'
+        pattern: '\[New Shit\] (.*) \[(.*)\] @ (https?\:\/\/.*\/).*id=(.*?) \[(.*?)\] ?(?:\[(.*?)\])?'
+        vars:
+          - torrentName
+          - category
+          - baseUrl
+          - torrentId
+          - torrentSize
+          - freeleech
+
+      - test:
+          - "[Scene] Uploaded 33 secs after pre"
+        pattern: '\[(Scene)\]\s+Uploaded (.*) after pre'
+        vars:
+          - scene
+          - preTime
+
+    match:
+      torrenturl: "/download.php?{{ .torrentId }}/{{ .passkey }}/{{ .torrentName }}.torrent"
```

Comment:
[This line in the .tracker file](https://github.com/autodl-community/autodl-trackers/blob/master/trackers/FunFile.tracker#L90) with `varenc` means it does a url encoding of the variable behind the scenes. They are using the torrent name in the url and it must be made sure to be safe for url use.

We support that by doing [it like here](https://github.com/autobrr/autobrr/blob/develop/internal/indexer/definitions/torrentleech.yaml#L72-L73).

```yml
torrenturl: "/download.php?{{ .torrentId }}/{{ .passkey }}/{{ .torrentName }}.torrent"
encode:
  - torrentName
```